worlds/gl/__init__.py

```diff
@@ -0,0 +1,288 @@
+            self.lock_item("Valley of Fire - Obelisk", "Mountain Obelisk 1")
+            self.lock_item("Dagger Peak - Obelisk", "Mountain Obelisk 2")
+            self.lock_item("Cliffs of Desolation - Obelisk", "Mountain Obelisk 3")
+            self.lock_item("Castle Courtyard - Obelisk", "Castle Obelisk 1")
+            self.lock_item("Dungeon of Torment - Obelisk", "Castle Obelisk 2")
+            self.lock_item("Poisoned Fields - Obelisk", "Town Obelisk 1")
+            self.lock_item("Haunted Cemetery - Obelisk", "Town Obelisk 2")
+
+        if not self.options.mirror_shards:
+            self.lock_item("Dragon's Lair - Dragon Mirror Shard", "Dragon Mirror Shard")
+            self.lock_item("Chimera's Keep - Chimera Mirror Shard", "Chimera Mirror Shard")
+            self.lock_item("Vat of the Plague Fiend - Plague Fiend Mirror Shard", "Plague Fiend Mirror Shard")
+            self.lock_item("Yeti's Cavern - Yeti Mirror Shard", "Yeti Mirror Shard")
+
+    def fill_slot_data(self) -> dict:
+        dshard = self.get_location("Dragon's Lair - Dragon Mirror Shard").item
+        yshard = self.get_location("Yeti's Cavern - Yeti Mirror Shard").item
+        cshard = self.get_location("Chimera's Keep - Chimera Mirror Shard").item
+        fshard = self.get_location("Vat of the Plague Fiend - Plague Fiend Mirror Shard").item
+        shard_values = [
+            item_dict[dshard.code] if dshard.player == self.player else [0x27, 0x4],
+            item_dict[yshard.code] if yshard.player == self.player else [0x27, 0x4],
+            item_dict[cshard.code] if cshard.player == self.player else [0x27, 0x4],
+            item_dict[fshard.code] if fshard.player == self.player else [0x27, 0x4],
+        ]
+        characters = [
+            self.options.unlock_character_one.value,
+            self.options.unlock_character_two.value,
+            self.options.unlock_character_three.value,
+            self.options.unlock_character_four.value,
+        ]
+        chests_barrels = self.options.chests_barrels.value
+        return {
+            "player": self.player,
+            "players": self.options.local_players.value,
+            "shards": shard_values,
+            "chests": bool(chests_barrels == 3 or chests_barrels == 1),
+            "barrels": bool(chests_barrels == 3 or chests_barrels == 2),
+            "speed": self.options.permanent_speed.value,
+            "keys": self.options.infinite_keys.value,
+            "characters": characters,
+            "max": (self.options.max_difficulty_value.value if self.options.max_difficulty_toggle else 4),
+            "instant_max": self.options.instant_max.value,
+            "death_link": bool((self.options.death_link.value == 1))
+        }
+
+    def create_items(self) -> None:
+        # First add in all progression and useful items
+        required_items = []
+        precollected = [item for item in item_list if item in self.multiworld.precollected_items[self.player]]
```

Comment:
Have you tested this `precollected` code?
It looks to me like `precollected` will always be empty, because `item` is an `ItemData`, and `self.multiworld.precollected_items` has `Item`, not `ItemData`